src/Microsoft.Diagnostics.Monitoring/DiagnosticsEventPipeProcessor.cs

```diff
@@ -289,6 +317,96 @@ private void PostMetric(Metric metric)
             }
         }
 
+        private Task HandleGCEvents(EventPipeEventSource source, int pid, Func<Task> stopFunc, CancellationToken token)
```

Comment:
At some point we do want to unify these anyway. I.e. there will be one implementation that collects gc dumps, and both the tool and dotnet-monitor use it.

Reply:
I think it makes more sense to put it into client SDK if that's the case, and make all the tools--`dotnet-monitor` included--be shells around the SDK.  At least from a code reuse and consistency standpoint.  Thoughts?  

CC @noahfalk @sywhang

Reply:
For reference: https://github.com/noahfalk/diagnostics/commit/2e546c49033d8e6a788c18246ba7c63a3091b2bd#commitcomment-38354471

Reply:
I'm not sure what code you are referring to when you say 'client SDK'?

Reply:
I meant the Microsoft.Diagnostics.NETCore.Client library that we package on NuGet.

Reply:
@josalem - My current suggestion is to unify them, but not in Microsoft.Diagnostics.NETCore.Client. I'd like to preserve M.D.N.C as a relatively thin library that makes diagnostic server calls whereas anything that contains the logic here pulls in a 2+ MB TraceEvent dependency. We probably wouldn't want +2MB dependency added to dotnet-dump because it needed to send one serialized message to the runtime. I sent a doc in email Friday that has a bit more detail about how I'd suggest factoring this, namely that this logic is in Microsoft.Diagnostics.Monitoring.EventPipe and both dotnet-gcdump and Microsoft.Diagnostics.Monitoring.RestServer would take a dependency on it.


@jander-msft + @wiktork - Is there urgency to get the gc-dump functionality checked in? It seems like we aren't that far from getting to the sharing model we are aiming for. I'm hoping to avoid creating/validating/bug fixing a different interim factoring of the logic if we can move along to the end-goal by doing something like this:
1) Refactor creating Microsoft.Diagnostics.Monitoring and Microsoft.Diagnostics.Monitoring.EventPipe.dll
2) Create the EventPipePipeline type with an API we think is about the right shape and have the existing RestServer counter/trace/logs APIs use it.
3) Move the GC dump support from dotnet-gcdump -> EventPipePipeline and make both dotnet-gcdump and the rest server consume it (similar to https://github.com/noahfalk/diagnostics/commit/2e546c49033d8e6a788c18246ba7c63a3091b2bd#commitcomment-38354471)